src/views/Rules/Rule/index.jsx

```diff
@@ -83,15 +86,27 @@ export default function Rule({ isNewRule, ...props }) {
   // 30 seconds - to make sure the helper text "Scheduled for ASAP" shows up
   const [scheduleDate, setScheduleDate] = useState(new Date());
   const [dateTimePickerError, setDateTimePickerError] = useState(null);
-  const [{ loading, error }, fetchRule] = useAction(getRule);
-  // eslint-disable-next-line no-unused-vars
-  const [scheduleChange, fetchScheduledChange] = useAction(getScheduledChange);
+  const [fetchRuleAction, fetchRule] = useAction(getRule);
+  const [scheduledChangeAction, fetchScheduledChange] = useAction(
+    getScheduledChange
+  );
+  const [addSCAction, addSC] = useAction(addScheduledChange);
+  const [updateSCAction, updateSC] = useAction(updateScheduledChange);
+  const [deleteSCAction, deleteSC] = useAction(deleteScheduledChange);
   const isLoading =
-    loading || scheduleChange.loading || products.loading || channels.loading;
+    fetchRuleAction.loading ||
+    scheduledChangeAction.loading ||
+    products.loading ||
+    channels.loading;
+  const error =
+    fetchRuleAction.error ||
+    scheduledChangeAction.error ||
+    addSCAction.error ||
+    updateSCAction.error ||
+    deleteSCAction.error;
```

Comment:
Whenever we are doing a mutation, we should be disabling the action buttons to let the user know an action has been triggered as well as stop the user from clicking the same button twice. We can track that in an `actionLoading` variable.

```jsx
const actionLoading = addSCAction.loading || updateSCAction.loading || deleteSCAction.loading;

// ...

{!isLoading && !actionLoading && (
    // action buttons
)}
```